Standards/scs-01XX-v1-secure-connections.md

```diff
@@ -0,0 +1,178 @@
+It also allows very little flexibility and might even make SCS conformance unappealing to CSPs due to the immense effort required to reach it.
+
+#### Option 2: only make recommendations
+
+This option would limit the SCS project to only recommend mechanisms in this standard like presented in the OpenStack Security Guide.
+Although this can increase awareness about the best practices recommended by OpenStack and maybe encourage CSPs to abide by them, it would actually contribute very little to the security baseline of SCS infrastructures as a whole since everything would stay optional.
+
+This option would be very easy to standardize and get consensus on due to its lightweight and optional nature.
+However, the actual added value for SCS is questionable at best.
+
+#### Option 3: mix recommendations and obligations
+
+This option forms a middle ground between options 1 and 2.
+For this, the standard needs to carefully assess each communication channel, mechanisms for protecting it and the effort required to do so as well as the implications.
+Then, only for mechanisms that are known to be reliable, are feasible to implement and for which the benefits clearly outweigh the integration effort required, should this standard enforce their implementation in a permissive way.
+For any remaining mechanisms the SCS standard should only make recommendations and refer to known best practices where applicable.
+
+This option would still offer improvements over arbitrary OpenStack clouds by establishing a baseline that goes beyond mere recommendations while still taking into account that not all communication channels are equally easy to secure and allowing necessary flexibility for the CSP.
+
+## Open questions
+
+### Verifying standard conformance for internal mechanisms
+
+Most of the mentioned communication channels to be secured are part of the internal IaaS infrastructure of a SCS cloud.
+When an internal protection mechanism is implemented by the CSP it cannot be verified from outside of the infrastructure without administrative access to the infrastructure.
+
+Thus, the SCS community is unable to fully assess a CSPs conformance to this standard without a dedicated audit of the infrastructure.
+
+## Decision
+
+This standard will mandate or recommend appropriate measures for securing the communication channels based on existing standards and recommendations.
+It will reference documents like the [OpenStack Security Guide](https://docs.openstack.org/security-guide/) where applicable.
+
+### Transport Layer Security (TLS)
+
+All server-side TLS configurations integrated into the infrastructure as covered by this standard MUST adhere to the folllowing rules:
+
+- The following deprecated TLS versions MUST NOT be used:
+  - v1.0
+  - v1.1
+- The following cipher suites MUST NOT be used:
+  - any RC4 cipher suite
+  - any DES cipher suite
+  - any cipher suite involving MD5 hash algorithms
+  - any cipher suite with less than 128-bit encryption
```

Comment:
That RFC about TLS deprecation is a good reference, I have added it to related documents for now.

On the general topic, I guess it's the same discussion as already happening in the issue and PR comments about those (admittedly not uncontroversial) rules currently in the standard.

When initially writing this section I wanted to start off with *something* in the standard to have base for discussion and also not go all-in mandating a very specific config for CSPs leaving no leeway. So I simply grabbed common (denylist) rules from ssl.com best practice guides, their referenced NIST standard and so on to get started.
I received a lot of valuable input on this so far, which I really appreciate. 

I think that most of this can be addressed by simply mandating conformance to an appropriate Mozilla recommendation[^1] instead. As far as I can tell, those are indeed allowlists, so there would be no "enumerating badness" problem.
I'd like to check the community's opinion on that in the upcoming IaaS call but if there's no veto I'd like to go with that.

[^1]: https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28recommended.29

Reply:
That sounds really good to me! For reference I hope I can merge the adjustments to the haproxy config in the reference implementation soon, so hopefully at least the haproxy part should be already compliant by then.